ouroboros-consensus/src/Ouroboros/Storage/ImmutableDB/Types.hs

```diff
@@ -75,13 +74,16 @@ extractTruncateFrom e = case e of
 -- @t@ (block). The returned 'SlotOffset's are from __first to last__ and
 -- __strictly__ monotonically increasing. The first 'SlotOffset' must be 0.
 --
+-- The @Maybe hash@ is the hash of the EBB, if present. The EBB itself should
+-- be the first entry in the list, if present.
+--
 -- We assume the output of 'EpochFileParser' to be correct, we will not
 -- validate it.
 --
 -- An error may be returned in the form of @'Maybe' e@. The 'SlotOffset's may
 -- be accompanied with other data of type @t@.
-newtype EpochFileParser e m t = EpochFileParser
-  { runEpochFileParser :: FsPath -> m ([(SlotOffset, t)], Maybe e) }
+newtype EpochFileParser e hash m t = EpochFileParser
```

Comment:
Let's document here what the precise behaviour is. In particular, why we are not reporting `Either e ..` but rather reporting the `e` as a separate `Maybe`, and how that relates to truncation.